src/torchio/data/image.py

```diff
@@ -561,6 +561,25 @@ def load(self) -> None:
         self.affine = affine
         self._loaded = True
 
+    def unload(self) -> None:
+        """Unload the image from memory.
+
+        Raises:
+            RuntimeError: If the images has not been loaded yet or if no path
+                is available.
+        """
+        if not self._loaded:
+            raise RuntimeError('Image cannot be unloaded as it is not loaded')
+        if self.path is None:
```

Comment:
For this to work nicer, maybe `self.path` should be updated on write (if still None at that point)?

Reply:
Do you mean setting `self.path` inside `self.save`? It sounds kind of nice, yes. But I feel like that's a bit too much magic and potentially unexpected behavior. I'm not sure about that feature.

Reply:
yes exactly as I find it weird if I saved the model and get the warning that there is no path present. Personally, I don't find it weird if the path wasn't set before

Reply:
That makes sense. But what if the path has been set before? I think there would be some uncertainty about the content of the `path` attribute depending on whether the image has been loaded or not. I mean: if you load image from path A and save to path B, would you set `image.path` to B too?

Reply:
Maybe that's a bad example. My worry is:

- Create image from tensor, save to path X. `image.path` is the **output** path
- Load image from path A, save to path B. `image.path` is the **input** path, whereas maybe one would expect it to be the output path

Reply:
I'm going to merge the PR so this is done, we can keep discussing it and revisit later. I hope that's ok.